### PR TITLE
Gauntlet Command Runner Helper

### PR DESCRIPTION
**Pull request by tateexon**
Helper functions for running gauntlet commands for solana and terra and future gauntlet e2e tests

### CODE REVIEW

File: gauntlet/gauntlet.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	log.Debug().Str("OS", runtime.GOOS).Msg("Runtime OS:")
+	version := "linux"
+	if runtime.GOOS == "darwin" {
+		version = "macos"
+	}
+	return version
+}
+
+// NewGauntlet Sets up a gauntlet struct and checks if the executable exists.
+func NewGauntlet(binPath string) (*Gauntlet, error) {
```
Comment (jleeh) [inline gauntlet/gauntlet.go:35]: Is `binPath` needed? Can `exec.LookPath("gauntlet")` be used?

**Review by jleeh — CHANGES_REQUESTED** [review]

File: gauntlet/gauntlet.go (hunk 1 above)
Comment (tateexon) [reply gauntlet/gauntlet.go:35]: The path and executable names are different between solana, terra, and older gauntlet 3. The executable will also rarely be in people's PATH variable.

Comment (tateexon) [reply gauntlet/gauntlet.go:35]: Although binPath is probably a poor name sine it needs to be the path and the executable itself. I will update the name.

File: gauntlet/gauntlet.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+// ExecCommandWithRetries Some commands are safe to retry and in ci this can be even more so needed.
+func (g *Gauntlet) ExecCommandWithRetries(args, errHandling []string, retryCount int) (string, error) {
+	var output string
+	var err error
+	for i := 0; i < retryCount; i++ {
+		log.Debug().Msg(fmt.Sprintf("Gauntlet Command Attempt: %v", i+1))
+		output, err = g.ExecCommand(args, errHandling)
+		if err != nil {
+			log.Warn().Str("Gauntlet Error", err.Error()).Msg("Failed Gauntlet Command Attempt")
+			time.Sleep(time.Second * 5)
```
[the remark this answers is not in the document]
Comment (tateexon) [reply gauntlet/gauntlet.go:116]: I forgot that existed, I will give it a go.

**Review by jleeh — APPROVED** [review]

**Review by kalverra — APPROVED** [review]

**Pull request merged**